driver/src/main/com/mongodb/client/gridfs/GridFSUploadStreamImpl.java

```diff
@@ -69,6 +69,15 @@ public ObjectId getFileId() {
     }
 
     @Override
+    public void abort() {
+        checkClosed();
```

Comment:
To avoid a data race on the closed field, the call to checkClosed should move into the synchronized block below.   In addition, checkClosed should synchronize on closeLock as well, as it's called in some cases outside synchronization on closeLock.